### DISCUSSION TITLE
an upstream response is buffered to a temporary file

### DISCUSSION THREAD

**Discussion by jpdsc** [NginxProxyManager/nginx-proxy-manager#1455] [Q&A]
Hello.

I was looking into the error logs and I see a lot of [warn] an upstream response is buffered to a temporary file.
I am running NPM on a Rasp PI 4 with 2GB ram.

Can I ignore these warning or is there something else I can do to resolve? I have read online and some state that it is due to low RAM to allocate the requests and therefore it's buffered to SD.

Thanks!

**Comment by wonder-wanderer** [comment]
can someone answer? 

this still happens in latest v2.11.2

**Comment by gizmocuz** [comment]
Same happens here.... No idea why this happens... downstream is a MediaWiki docker compose project.
I also get "upstream prematurely closed connection while reading response header from upstream" errors when I go to the Mediawiki preferences

**Comment by mordyovits** [comment]
The proxied backend service is on the same machine, so the nginx proxy is fed the response far faster than the client on the internet can read it. Nginx frees up the backend by reading to the end of the response and buffering it to disk. It then serves the slower client from the buffered data. Config determines when it buffers responses like this. 

I have no need of this feature, so I disabled it by adding 'proxy_buffering off;' to the custom options of the NPM proxy host (in the advanced tab). That ended those [warn] log entries.
